test/common/grpc/grpc_client_integration_test.cc

```diff
@@ -29,6 +29,35 @@ TEST_P(GrpcClientIntegrationTest, BasicStream) {
   dispatcher_helper_.runDispatcher();
 }
 
+// Validate that a simple request-reply stream works with bytes metering.
+TEST_P(GrpcClientIntegrationTest, BasicStreamWithBytesMeter) {
+  // Currently, bytes metering is only implemented in Envoy gRPC. Therefore, skip this test for
+  // google gRPC.
+  SKIP_IF_GRPC_CLIENT(ClientType::GoogleGrpc);
+  initialize();
+  auto stream = createStream(empty_metadata_);
+
+  // Create the send request.
+  helloworld::HelloRequest request_msg;
+  request_msg.set_name(HELLO_REQUEST);
+
+  RequestArgs request_args;
+  request_args.request = &request_msg;
+  stream->sendRequest(request_args);
+  stream->sendServerInitialMetadata(empty_metadata_);
+
+  auto send_buf = Common::serializeMessage(request_msg);
+  Common::prependGrpcFrameHeader(*send_buf);
+  // Verify that the number of sent bytes that is tracked in stream info equals to the length of
```

Comment:
Do we have any tests that account for multi-part streams?

Reply:
I just added `MultiStreamWithBytesMeter` test.